File: app/src/main/java/com/jermaine/awesomelibraries/view/activity/additem/AddRepoContract.java
package com.jermaine.awesomelibraries.view.activity.additem;


import com.jermaine.awesomelibraries.base.BasePresenter;
import com.jermaine.awesomelibraries.base.BaseView;

public interface AddRepoContract {
    interface AddRepoPresenter extends BasePresenter {
        void onAddClick();
    }

    interface AddRepoView extends BaseView {
        String getId();

        String getName();

        String getDescription();

        void setIdError();

        void setNameError();

        void setDescriptionError();

        void finishScreen();
    }
}

File: app/src/main/java/com/jermaine/awesomelibraries/uicontroller/activity/additem/AddRepoContract.kt
package com.jermaine.awesomelibraries.uicontroller.activity.additem


import com.jermaine.awesomelibraries.base.BaseView

interface AddRepoContract {
    interface AddRepoView : BaseView {
        val id: String

        val name: String

        val description: String

        fun setIdError()

        fun setNameError()

        fun setDescriptionError()

        fun finishScreen()
    }
}
